mlx/primitives.cpp

```diff
@@ -3950,4 +3950,37 @@ bool View::is_equivalent(const Primitive& other) const {
   return (dtype_ == a_other.dtype_);
 }
 
+std::pair<std::vector<array>, std::vector<int>> Hadamard::vmap(
```

Comment:
```suggestion
bool Hadamard::is_equivalent(const Primitive& other) const {
  const Hadamard& h_other = static_cast<const Hadamard&>(other);
  return scale_ == h_other.scale_;
}

std::pair<std::vector<array>, std::vector<int>> Hadamard::vmap(
```

Reply:
Also needs the declaration in `primitives.h` of course.

Reply:
Thanks! Added this in the latest commit.